src/ir_Bosch.cpp

```diff
@@ -25,37 +27,298 @@ const uint16_t kBoschZeroSpace = 571;
 const uint16_t kBoschFreq = 38000;  // Hz. (Guessing the most common frequency.)
 const uint16_t kBosch144NrOfSections = 3;
+const uint16_t kBosch144BytesPerSection = 6;
+
+using irutils::addBoolToString;
+using irutils::addModeToString;
+using irutils::addFanToString;
+using irutils::addTempToString;
 
 #if SEND_BOSCH144
-/// Send a Bosch 144-bit / 18-byte message
-/// Status: STABLE / Confirmed Working.
+/// Send a Bosch 144-bit / 18-byte message ( 96-bit message are also possible)
+/// Status: BETA / Probably Working.
 /// @param[in] data The message to be sent.
 /// @param[in] nbytes The number of bytes of message to be sent.
 /// @param[in] repeat The number of times the command is to be repeated.
-void IRsend::sendBosch144(const unsigned char data[], const uint16_t nbytes,
-                          const uint16_t repeat) {
-  // nbytes is required to be a multiple of kBosch144NrOfSections.
-  if (nbytes % kBosch144NrOfSections != 0) return;
-
+void IRsend::sendBosch144( const unsigned char data[], const uint16_t nbytes,
+                          const uint16_t repeat )
+{
+  // nbytes is required to be a multiple of kBosch144BytesPerSection.
+  if( nbytes % kBosch144BytesPerSection != 0 )return;
   // Set IR carrier frequency
-  enableIROut(kBoschFreq);
+  enableIROut( kBoschFreq );
 
-  for (uint16_t r = 0; r <= repeat; r++) {
-    const uint16_t kSectionByteSize = nbytes / kBosch144NrOfSections;
-    for (uint16_t offset = 0; offset < nbytes; offset += kSectionByteSize)
+  for( uint16_t r = 0; r <= repeat; r++ )
+  {
+    for( uint16_t offset = 0; offset < nbytes; offset += kBosch144BytesPerSection )
       // Section Header + Data + Footer
-      sendGeneric(kBoschHdrMark, kBoschHdrSpace,
-                  kBoschBitMark, kBoschOneSpace,
-                  kBoschBitMark, kBoschZeroSpace,
-                  kBoschBitMark, kBoschFooterSpace,
-                  data + offset, kSectionByteSize,
-                  kBoschFreq, true, 0, kDutyDefault);
-    space(kDefaultMessageGap);  // Complete guess
+      sendGeneric( kBoschHdrMark, kBoschHdrSpace,
+                   kBoschBitMark, kBoschOneSpace,
+                   kBoschBitMark, kBoschZeroSpace,
+                   kBoschBitMark, kBoschFooterSpace,
+                   data + offset, kBosch144BytesPerSection,
+                   kBoschFreq, true, 0, kDutyDefault );
+    space( kDefaultMessageGap );  // Complete guess
   }
 }
 
 #endif  // SEND_BOSCH144
 
+/// Class constructor.
+/// @param[in] pin GPIO to be used when sending.
+/// @param[in] inverted Is the output signal to be inverted?
+/// @param[in] use_modulation Is frequency modulation to be used?
+IRBosch144AC::IRBosch144AC( const uint16_t pin, const bool inverted, const bool use_modulation )
+    : _irsend( pin, inverted, use_modulation )
+    {
+      stateReset();
+    }
+
+/// Reset the internal state to a fixed known good state.
+void IRBosch144AC::stateReset( void ) 
+{
+  setRaw( kBosch144DefaultState );
```

Comment:
Also set an intial value for `powerFlag` too.

Reply:
is now set to "true" there.